hadoop-tools/hadoop-distcp/src/test/java/org/apache/hadoop/tools/mapred/TestCopyCommitter.java

```diff
@@ -177,6 +177,44 @@ public void testPreserveStatus() throws IOException {
       conf.unset(DistCpConstants.CONF_LABEL_PRESERVE_STATUS);
     }
 
+  }
+  @Test
+  public void testPreserveStatusWithAtomicCommit() throws IOException {
+    TaskAttemptContext taskAttemptContext = getTaskAttemptContext(config);
+    JobContext jobContext = new JobContextImpl(taskAttemptContext.getConfiguration(),
+                            taskAttemptContext.getTaskAttemptID().getJobID());
+    Configuration conf = jobContext.getConfiguration();
+    String sourceBase;
+    String workBase;
+    String targetBase;
+    FileSystem fs = null;
+    try {
+      OutputCommitter committer = new CopyCommitter(null, taskAttemptContext);
+      fs = FileSystem.get(conf);
```

Comment:
this wil be the HDFS minicluster, right?

Reply:
Yeah, we have registered a HDFS minicluster.

Reply:
Can we get the code merged?

Reply:
@steveloughran Can you look at this PR. The test case seems sufficient for testing this use case.

Reply:
@steveloughran Can this code be merged?